actionpack/lib/action_dispatch/middleware/exception_wrapper.rb

```diff
@@ -174,6 +174,31 @@ def self.status_code_for_exception(class_name)
       Rack::Utils.status_code(@@rescue_responses[class_name])
     end
 
+    def show?(request)
+      # We're treating `nil` as "unset", and we want the default setting to be
+      # `:all`. This logic should be extracted to `env_config` and calculated
+      # once.
+      config = request.get_header("action_dispatch.show_exceptions")
+
+      # For backwards compatibility.
+      if config == true
+        ActiveSupport.deprecator.warn("Setting action_dispatch.show_exceptions to true is deprecated. Set to :all instead.")
```

Comment:
Is it possible to add a test for the deprecations or did I miss them?

Reply:
Good call. I added tests in `actionpack/test/dispatch/exception_wrapper_test.rb` to demonstrate the new deprecation warning.

Reply:
Also I think we will want to use `ActionDispatch.deprecator` here, see #47354